plugins/psv-extractor/src/psv.extractor.spec.ts

```diff
@@ -1,56 +1,89 @@
+import * as path from 'path'
 
-jest.mock("fs");
+describe('PsvParser', function () {
+  const buffer: Buffer = fs.readFileSync(
+    path.join(__dirname, '../ref/test-basic.psv')
+  )
 
-describe("PsvParser", function () {
   const parser = new PsvExtractor({
-    id: "some_id",
+    id: 'some_id',
     topic: Flatfile.EventTopic.FileCreated,
-    payload: {},
+    payload: {} as Record<string, unknown>,
     createdAt: new Date(),
-    domain: "space",
+    domain: 'space',
+    name: 'upload:completed',
     context: {
-      fileId: "dev_fl_dZNtPPTa",
-      spaceId: "dev_sp_w2TZIUBE",
-      accountId: "dev_acc_Iafc9fLm",
-      environmentId: "dev_env_rH3SeKkh",
-    },
-  });
-
-  describe("test-basic.psv", function () {
-    beforeEach(() => {
-      const buffer = Buffer.from(
-        '"Code"|"Details"|"BranchName"|"Tenant"\n"Personal Care"|"Personal Care"|"Department"|"notdata"\n"Home Nursing"|"Home Nursing"|"Department"|"notdata"',
-        "utf8"
-      );
-      jest.spyOn(fs, "readFileSync").mockReturnValue(buffer);
-    });
-
-    test("finds the sheet name", () => {
-      const capture = parser.parseBuffer(
-        '"Code"|"Details"|"BranchName"|"Tenant"\n"Personal Care"|"Personal Care"|"Department"|"notdata"\n"Home Nursing"|"Home Nursing"|"Department"|"notdata"'
-      );
-      const sheetNames = capture ? Object.keys(capture) : [];
-      expect(sheetNames).toEqual(["Sheet1"]);
-    });
-
-    test("finds the header names", () => {
-      const capture = parser.parseBuffer(
-        '"Code"|"Details"|"BranchName"|"Tenant"\n"Personal Care"|"Personal Care"|"Department"|"notdata"\n"Home Nursing"|"Home Nursing"|"Department"|"notdata"'
-      );
-      const headers =
-        capture && capture["Sheet1"] ? capture["Sheet1"].headers : [];
-      expect(headers).toHaveLength(4); // Assuming there are 4 headers
-    });
-
-    test("finds values", () => {
-      const capture = parser.parseBuffer(
-        '"Code"|"Details"|"BranchName"|"Tenant"\n"Personal Care"|"Personal Care"|"Department"|"notdata"\n"Home Nursing"|"Home Nursing"|"Department"|"notdata"'
-      );
-      const data = capture && capture["Sheet1"] ? capture["Sheet1"].data : [];
-      expect(data.length).toBeGreaterThan(0); // Assuming there's at least one row
-    });
-  });
-});
+      fileId: 'dev_fl_dZNtPPTa',
+      spaceId: 'dev_sp_w2TZIUBE',
+      accountId: 'dev_acc_Iafc9fLm',
+      environmentId: 'dev_env_rH3SeKkh',
+    } as any,
+    api: {} as any,
+  } as Flatfile.UploadCompletedEvent)
+
+  describe('test-basic.psv', function () {
+    test('finds the sheet name', () => {
+      const capture = parser.parseBuffer(buffer)
+      const sheetNames = Object.keys(capture)
+      expect(sheetNames).toEqual(['Sheet1'])
+    })
+
+    test('finds the header names', () => {
+      const capture = parser.parseBuffer(buffer)
+      const headers = capture['Sheet1'].headers
+      expect(headers).toHaveLength(4) // Assuming there are 4 headers
```

Comment:
⛳ This comment doesn't provide much value beyond what the code reads. Maybe we should remove it.